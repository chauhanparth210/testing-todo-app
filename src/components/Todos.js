import React from "react";

export const Todos = ({ todos }) => {
  return (
    <ul className="todos">
      {(todos || []).map((todo, index) => (
        <li key={index}>{todo}</li>
      ))}
    </ul>
  );
};
