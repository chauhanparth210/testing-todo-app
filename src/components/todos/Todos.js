import React from "react";

export const Todos = ({ todos, removeTodo }) => {
  return (
    <ul className="todos">
      {(todos || []).map((todo, index) => (
        <li key={index}>
          {todo}
          <button
            style={{
              marginLeft: "8px",
            }}
            onClick={() => removeTodo(todo)}
          >
            x
          </button>
        </li>
      ))}
    </ul>
  );
};
