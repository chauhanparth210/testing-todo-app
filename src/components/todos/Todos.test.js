import React from "react";
import { shallow } from "enzyme";
import TestRenderer from 'react-test-renderer';
import { Todos } from "./Todos";

describe("Todos component testing", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Todos debug />);
    expect(wrapper.exists()).toBeTruthy();
  });

  it("should render Todos component properly", () => {
    const testRenderer = TestRenderer.create(<Todos debug />);
    expect(testRenderer).toMatchSnapshot();
  });

  it("should render Todos component properly with todos prop",()=>{
    const todos = ["hello","I need to poo ðŸ’©"]
     wrapper = TestRenderer.create(<Todos todos={todos} />);  
    expect(wrapper).toMatchSnapshot();
  })

  // it("initially todos should empyt", () => {
  //   let todosListElmt = wrapper.find(".todos");
  //   expect(todosListElmt.find("ul").children()).toHaveLength(0);
  // });

  // it("shoudl render todos", () => {
  //   const todos = ["hello", "testing"];
  //   wrapper = shallow(<Todos todos={todos} />);
  //   let todosListElmt = wrapper.find(".todos");
  //   expect(todosListElmt.find("ul").children()).toHaveLength(todos.length);
  // });

  // it("pass todo to main state", () => {
  //   const todos = ["hello"];
  //   const removeTodoMockFn = jest.fn();
  //   wrapper = shallow(<Todos todos={todos} removeTodo={removeTodoMockFn} />);
  //   let todosListElmt = wrapper.find(".todos");
  //   expect(todosListElmt.find("ul").children()).toHaveLength(todos.length);

  //   expect(removeTodoMockFn).not.toBeCalled();

  //   let deletedTodoRemoveBtnElement = wrapper.find(".todos>li>button");
  //   deletedTodoRemoveBtnElement.simulate("click");

  //   jest.useFakeTimers();

  //   setTimeout(() => {
  //     expect(removeTodoMockFn).toBeCalled();
  //     expect(removeTodoMockFn).toHaveBeenCalledTimes(1);
  //   }, 1000);
  // });
});
