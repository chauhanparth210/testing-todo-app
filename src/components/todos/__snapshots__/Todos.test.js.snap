// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todos component testing should render Todos component properly 1`] = `
<ul
  className="todos"
/>
`;

exports[`Todos component testing should render Todos component properly with todos prop 1`] = `
<ul
  className="todos"
>
  <li>
    hello
    <button
      onClick={[Function]}
      style={
        Object {
          "marginLeft": "8px",
        }
      }
    >
      x
    </button>
  </li>
  <li>
    I need to poo ðŸ’©
    <button
      onClick={[Function]}
      style={
        Object {
          "marginLeft": "8px",
        }
      }
    >
      x
    </button>
  </li>
</ul>
`;
